google/cloud/security/scanner/scanners/instance_network_interface_scanner.py

```diff
@@ -0,0 +1,122 @@
+# you may not use this file except in compliance with azthe License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Scanner for the Networks Enforcer acls rules engine."""
+
+# pylint: disable=line-too-long
+from google.cloud.security.common.util import log_util
+from google.cloud.security.common.data_access import instance_dao
+from google.cloud.security.common.data_access import project_dao
+from google.cloud.security.common.gcp_type.resource import ResourceType
+from google.cloud.security.scanner.scanners import base_scanner
+# pylint: enable=line-too-long
+
+LOGGER = log_util.get_logger(__name__)
+
+
+class InstanceNetworkInterfaceScanner(base_scanner.BaseScanner):
+    """Pipeline to network enforcer from DAO."""
+
+    def __init__(self, global_configs, snapshot_timestamp):
+        """Initialization.
+
+        Args:
+            snapshot_timestamp: The snapshot timestamp
+        """
+        super(InstanceNetworkInterfaceScanner, self).__init__(
+            global_configs,
+            snapshot_timestamp)
+        self.snapshot_timestamp = snapshot_timestamp
+
+    def get_instance_networks_interfaces(self):
+        """Get network info from a particular snapshot.
+
+           Returns:
+               A list of networks from a particular project
+           Raises:
+               MySQLError if a MySQL error occurs.
+        I set the list  to get rid of duplicates
+        """
+        instances = instance_dao.InstanceDao().get_instances(
+            self.snapshot_timestamp)
+        network_interfaces = []
+        for instance in instances:
+            network_interfaces += instance.create_network_interfaces()
+        return network_interfaces
+
+    def parse_instance_network_instance(self, instance_object):
+        """Create a list of network interface obj."""
+        return instance_object.create_network_interfaces()
+
+    def _get_project_policies(self):
+        """Get projects from data source."""
+        project_policies = {}
+        project_policies = (
+            project_dao.ProjectDao().get_project_policies('projects',
+                                                          self.
+                                                          snapshot_timestamp))
+        return project_policies
+
+    @staticmethod
+    def _get_resource_count(project_policies, instance_network_interfaces):
+        """Get resource count for org and project policies.
+
+        Args:
+            org_policies: organization policies from inventory.
+            project_policies: project policies from inventory.
```

Comment:
it looks like the Args need to be updated here.

Also wondering, do you mean to get the projects, or the project IAM policies?